### ISSUE TITLE
NV56TU code does not build (reported in DM by Theirry)

### ISSUE THREAD

**Issue by pietrushnic** [Dasharo/dasharo-issues#1139]
### Component

Dasharo firmware

### Device

NovaCustom V56 14th Gen

### Dasharo version

v0.9.0

### Brief summary

Commit reference config points to bad FSP paths, no recipe to build EC

### How reproducible

I'm unsure; I'm just reporting what Thierry sent me in DM. I assume it should 100%.

https://github.com/linuxboot/heads/issues/1835

### How to reproduce

Try to build https://docs.dasharo.com/variants/novacustom_v560tu/releases/#v090-2024-07-18

### Expected behavior

Dasharo (coreboot+UEFI) builds correctly based on the provided information in the release.

### Actual behavior

No sure, no logs, no error messages received. Someone has to try to reproduce that and confirm this is valid.I'm not sure. No logs, no error messages were 

### Additional context

It blocks Heads development. From Thierry:

> * reference config does not point to proper FSP header and other paths are wrong (untested?)
> * FMP will need to be redefined since there is no contiguous free space in firmware (we need 7.5mb for payload in CBFS, and all other free space), never had to write up/modify FMP before where ifd in tree cannot be parsed with ifdtool
> * ec is not built by Heads, need code there under modules/coreboot as contribution (patch)

> I see https://github.com/linuxboot/heads/pull/1846 changing used coreboot fork commit to accomodate same kind of issues discovered by mkopec.

**Comment by tlaurion** [comment]
To validate this, defconfig needs to be expended into ooldconfig. Here I can only refer to what is under defconfig of ref config  at
https://github.com/Dasharo/coreboot/blob/316f964cb0f892ad4fe985b16c598f1e5ed8fe61/configs/config.novacustom_v560tu

- FSP paths not existing: https://github.com/Dasharo/coreboot/blob/316f964cb0f892ad4fe985b16c598f1e5ed8fe61/configs/config.novacustom_v560tu#L17-L18
- EC firmware. How to build under Heads to point to it for inclusion?
- CBFS_SIZE + IFD/FIT tables need to work with CBFS >= 8MB

TLDR: code, paths + reference config needs to build under Heads and then Heads features tested on my side under https://github.com/linuxboot/heads/issues/1835

**Comment by tlaurion** [comment]
Pertinent references of PR that led to switch from i915 to efifb with coreboot prepared FB being compliant to efifb, needed for both V560TU and V540TU (meteorlake) at https://github.com/linuxboot/heads/issues/1835#issuecomment-2479308376

**Comment by macpijan** [comment]
@mkopec is it already resolved in: https://github.com/linuxboot/heads/pull/1846 ?
Can you please clarify what was the problem here?

**Comment by mkopec** [comment]
> FSP paths
Needed to change path from novacustom-blobs (private) to dasharo-blobs (public). The FSPs themselves are the same.

> EC firmware
Probably need to add a new step in CI, but this is the first release, so I'm not sure if EC FW needs to be included

> CBFS_SIZE + IFD/FIT tables need to work with CBFS >= 8MB so payload can be stitched in.
Not sure about this one, so far I've managed to build coreboot with LPSS UART for debugging. coreboot logs end with jumping to payload, but nothing happens afterwards (on console or on screen)

**Comment by tlaurion** [comment]
https://github.com/linuxboot/heads/pull/1846#issuecomment-2486040508

**Comment by macpijan** [comment]
@mkopec I think you have not responded to me previous message in full. Can we please point to repo revision which can be built by all users? We are getting another reports via Matrix.

**Comment by macpijan** [comment]
We need clear information on what's the problem here and link / inform in the documentation which tags cannot be build, and whether there are some workarounds we want propose.

**Comment by tlaurion** [comment]
> We need clear information on what's the problem here and link / inform in the documentation which tags cannot be build, and whether there are some workarounds we want propose.

Its simply missing https://github.com/linuxboot/heads/pull/1846#issuecomment-2512333110 so collaboration that happened there is merged into dasharo, builds ans I can continue my collaboration under v560tu to do the same thing I did for V540TU, but own a v560tu so _should_ land faster from my side since I will be able to test and unbrick if it builds and doesn't cause a brick.


I still have no news on EC being built and integrated into coreboot stitched rom to self sync ec, since that wasn't part of V540TU AFAIK.

**Comment by mkopec** [comment]
@macpijan I fixed the build issue for Heads, but for UEFI it's a different matter, because there we depend on the Intel GOP driver which AIUI cannot be published standalone, unincorporated into a binary

**Comment by Algebro7** [comment]
Has there been any progress on resolving this issue for UEFI? The public facing docs still say that it's possible to build it, and without being able to do so there is no way to verify the integrity of the firmware. Thanks!

**Comment by mkopec** [comment]
@Algebro7 We have a new build system, please check out https://github.com/Dasharo/fwbuild and let us know if it works for you :)

This release was quite problematic due to blobs and their licensing, this system works around these issues

**Comment by Algebro7** [comment]
Nice! I see a FWBUILD for the v540 in there, will that work for the v560?

**Comment by mkopec** [comment]
@Algebro7 Please check again, I just pushed V560TU too

**Comment by Algebro7** [comment]
Alright, it built, will try flashing it later. BTW, what is this blob here? https://github.com/Dasharo/fwbuild/blob/main/v560tu/uefi/v0.9.0/bios/FWBUILD#L35

**Comment by mkopec** [comment]
That is the intel gigabit ethernet driver, it's needed for network boot support. It may only be redistributed when integrated into a full firmware binary, so we had to add this workaround

**Comment by Algebro7** [comment]
Thanks! I tried flashing the ec rom from DTS today and it's failing with the following errors:

```
# flashrom -p ite_ec:boardmismatch=force,romsize=128K -w novacustom_v560tu_ec.rom
ITE EC programmer not (yet) supported on this device
Proceeding anyway because user forced us to
Failed to read EC project
Failed to read EC version
Flash Part ID: 00 00 00
Found ENE flash chip "KB9012 (EDI)" (128 kB)
Found Programmer flash chip "Opaque flash chip" (128 kB, Programmer-specific) on ite_ec.
Reading old flash chip contents... Flash read failed @ 0x0
FAILED.
```

Any idea what's up with that? There were no errors/warnings in the build process.

**Comment by mkopec** [comment]
If you already have Dasharo EC installed you can use `dasharo_ectool flash` to update your EC firmware.

**Comment by Algebro7** [comment]
And how can I troubleshoot if I don't? I don't necessarily trust the firmware that was pre-flashed on this device so I'd like to make sure it's possible "start fresh".

**Comment by mkopec** [comment]
If you had Clevo proprietary EC firmware, then you would use `flashrom -p ite_ec` like you tried here: https://github.com/Dasharo/dasharo-issues/issues/1139#issuecomment-2543379913

The flashing protocol is different depending on which EC firmware you have. While Clevo EC firmware is running, you need to use flashrom, but while Dasharo EC is running, you would need to use dasharo_ectool.

**Comment by Algebro7** [comment]
Ah OK, thanks for the explanation. `dasharo_ectool` seems to have worked fine :)

**Comment by tlaurion** [comment]
https://github.com/linuxboot/heads/pull/1846#issuecomment-2555027475
